src/components/Form/Widgets/PasswordWidget.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+	Box,
+	FormControl,
+	IconButton,
+	InputAdornment,
+	TextField,
+	Typography,
+} from '@mui/material';
+import * as React from 'react';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { WidgetProps } from '@rjsf/utils';
+import zxcvbn from 'zxcvbn';
+
+const STRENGTH_TITLES = ['Very weak', 'Weak', 'Good', 'Strong', 'Very strong'];
+const STRENGTH_STYLES = [
+	{ width: 0 },
+	{ width: '25%', backgroundColor: 'orange' },
+	{ width: '50%', backgroundColor: 'yellow' },
+	{ width: '75%', backgroundColor: 'green' },
+	{ width: '100%', backgroundColor: 'darkgreen' },
+];
+
+export interface PasswordStrengthProps {
+	password?: string;
+}
+
+const PasswordStrength = ({ password }: PasswordStrengthProps) => {
+	const [strengthScore, setStrengthScore] = React.useState<
+		number | undefined
+	>();
+
+	React.useEffect(() => {
+		if (password) {
+			try {
+				const { score } = zxcvbn(password);
+				setStrengthScore(score);
+			} catch {
+				// Ignore any errors, as we only want to show the strength meter if it is available.
+			}
+		}
+	}, [password]);
+
+	if (!password || strengthScore === undefined) {
+		return null;
+	}
+
+	return (
+		<Box>
+			<Typography>
+				Password strength: <em>{STRENGTH_TITLES[strengthScore]}</em>
+			</Typography>
+			<Box
+				sx={[
+					STRENGTH_STYLES[strengthScore],
+					{ height: '4px', transition: 'all ease-out 150ms' },
+				]}
+			/>
+		</Box>
+	);
+};
+
+export const PasswordWidget = ({
+	id,
+	label,
+	value,
+	disabled,
+	name,
+	placeholder,
+	onBlur,
+	onChange,
+	onFocus,
+	options,
+	required,
+}: WidgetProps) => {
+	const [showPassword, setShowPassword] = React.useState(false);
+
+	const change = ({ target: { value } }: any) => {
+		return onChange(value === '' ? options.emptyValue : value);
+	};
+
+	return (
+		<FormControl>
+			<TextField
+				label={label}
+				required={required}
+				id={id}
+				type={showPassword ? 'text' : 'password'}
```

Comment:
If the user submits with the text field type as `text`, will their browser still offer to save the password for them? If not, then we should not be doing this and there should be a way to make this work with just `type='password'`

Reply:
@myarmolinsky it was as you said, but modern browsers and password managers often have auto-fill features that can recognise form fields based on the name attribute. 
The approch is pretty standard, the only way to prevent this behaviour today is to have autocomplete="off" which we don't use